website/content/docs/agent/template-config.mdx

```diff
@@ -0,0 +1,48 @@
+---
+layout: docs
+page_title: Vault Agent Template Config
+description: |-
+  Vault Agent's Template Config to set Templating Engine behavior
+---
+
+# Vault Agent Template Config
+
+Template Config allows users of Vault Agent to determine the behavior that
+pertains to the templating engine.
+
+## Functionality
+
+The `template_config` stanza. Note that `template_config` can only be defined
+once, and is different from the `template` stanza. Unlike `template` which
+focuses on where and how a specific secret is rendered, `template_config`
+contains parameters affects how the templating engine as a whole behaves and its
+interaction with the rest of Agent. This includes, but is not limited to,
+program exit behavior. Other parameters that applies to the templating engine as
+a whole may be added over time.
+
+### Interaction with `error_on_missing_key`
+
+The parameter `error_on_missing_key` can be specified within the `template`
+stanza which determines if a template should ignore rendering a missing key
+within the response of a secret or error out. When `error_on_missing_key` is
+not specified or set to `false` and the key to render is not in the secret's
+response, the templating engine will ignore it (or render `"<no value>"`)
+and continue on with its rendering.
+
+If the desire is to have Agent fail and exit on a missing key, both
+`template.error_on_missing_key` and `template_config.exit_on_retry_failure` must
+be set to true. Otherwise, the templating engine will error and render to its
+destination, but agent will not exit and retry until it's able to do so or until
+the process is terminated.
+
+Note that a missing key from a secret's response is different from a missing or
+non-existent secret. The former's behavior can be toggled through
+`error_on_missing_key`, while the latter always results in an error returned by
+the templating engine.
```

Comment:
```suggestion
non-existent secret. The templating engine will always error if a secret is missing, but will only error for a missing key if `error_on_missing_key` is set. Whether Vault Agent will exit when the templating engine errors depends on the value of `exit_on_retry_failure`.
```

I had a quick go at reworking this because I found it a bit confusing. It sounded a bit like "always results in an error" means `exit_on_retry_failure` might be ignored in some cases.

Reply:
Yeah this note was meant purely for `error_on_missing_key` as there was some misunderstanding of how it worked for me in the past. I like how you tie it back to `exit_on_retry_failure` though, so I'll go ahead and accept your suggestion!